buildSrc/src/main/groovy/io/deephaven/project/Type.groovy

```diff
@@ -0,0 +1,89 @@
+package io.deephaven.project
+
+import groovy.transform.CompileStatic
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.artifacts.ProjectDependency
+import org.gradle.api.plugins.JavaPlugin
+import org.gradle.api.tasks.TaskProvider
+import org.gradle.language.base.plugins.LifecycleBasePlugin
+
+@CompileStatic
+class Type {
```

Comment:
I think `Type` is one of the least descriptive name possible...  what _type_ of Type is it? ^-^